### PR TITLE
Custom Promise Types: Improvements made while working on blog posts

### PR DESCRIPTION
**Pull request by olehermanse**

### CODE REVIEW

File: docs/custom_promise_types/modules.md [hunk 1]
```diff
@@ -120,10 +120,21 @@ In short, when writing a promise module, these log levels should be used:
...
+* `warning` - The promise did not fail, but there is something the user (policy writer) should probably fix. Some examples:
+  * Policy relies on deprecated behavior/syntax which will change
+  * Policy uses demo / unsafe options which should be avoided in a production environment
+* `notice` - Unusual events which you want to notify the user about
+  * Most promise types won't need this - usually `info` or `warning` is more appropriate
+  * Useful for events which happen rarely and are not the result of a promise, for example:
+    * New credentials detected
+    * New host bootstrapped
+    * The module made a change to the system for itself to work (database initialized, user created)
+* `info` - Changes made to the system (1 per repaired promise)
```
Comment (vpodzime) [inline docs/custom_promise_types/modules.md:132]: Why *1 per repaired promise*?

File: docs/custom_promise_types/modules.md [hunk 2]
```diff
@@ -120,10 +120,21 @@ In short, when writing a promise module, these log levels should be used:
...
 * `verbose` - Human understandable detailed information about promise evaluation
 * `debug` - Programmer-level information that is only useful for CFEngine developers or module developers
 
+Note that all log levels, except for `debug`, should be friendly to non-developers, and not include programmer's details (such as protocol messages, source code references, function names, etc.).
```
Comment (vpodzime) [inline docs/custom_promise_types/modules.md:136]: I'd prefer this line to be wrapped/split. There are other long lines in the file so maybe one more commit wrapping them?

**Review by vpodzime — COMMENTED** [review]
Looks good to me otherwise.

File: docs/custom_promise_types/modules.md (hunk 2 above)
Comment (olehermanse) [reply docs/custom_promise_types/modules.md:136]: I always put one sentence per line in markdown, since it makes the diffs very nice.

Comment (olehermanse) [reply docs/custom_promise_types/modules.md:136]: (and it doesn't matter when you read the rendered version)

File: docs/custom_promise_types/modules.md (hunk 1 above)
Comment (olehermanse) [reply docs/custom_promise_types/modules.md:132]: We could add a "usually" or something. It's a good rule to follow in general. 0 is too few 2 is too many :)

Comment (vpodzime) [reply docs/custom_promise_types/modules.md:132]: I can imagine the promises handled by promise modules more than any other promises will be making multiple changes at once.

Comment (olehermanse) [reply docs/custom_promise_types/modules.md]: Added a clarifying sentence: https://github.com/cfengine/core/pull/4476/commits/13f4d84033fad96f2d23e9bbfa6485a4623bd61e

**Review by vpodzime — APPROVED** [review]
ACK

**Pull request merged**